deployment/README.md

```diff
@@ -0,0 +1,41 @@
+# Trillian supported deployments
+
+This directory contains the supported deployment types for Trillian. Currently,
+this includes:
+
+- [Kubernetes on GCP using Cloud
+  Spanner](#kubernetes-on-gcp-using-cloud-spanner)
+
+Further deployment types (community-supported) can be found in the
+examples/deployment directory.
+
+# Kubernetes on GCP using Cloud Spanner
+
+## Architecture
+
+The infrastructure is created using Terraform. It consists of a Cloud Spanner
+database, a Trillian-specific etcd cluster, the Trillian logservice and
+logsigner. A Workload Identity is setup by Terraform to give the services access
+to Cloud Spanner. The container images are the latest official release.
```

Comment:
"the latest official release" - this could easily become inaccurate if someone creates a new release but doesn't update this doc. Probably better to just say "v1.3.3" here.